addons/mail/static/src/core/common/message_model.js

```diff
@@ -100,54 +80,51 @@ export class Message extends Record {
                 partner: this._store.self,
             };
         }
-        if ("messageReactionGroups" in data) {
-            this.reactions = data.messageReactionGroups;
-        }
         if (this.isNotification && !this.notificationType) {
             const parser = new DOMParser();
             const htmlBody = parser.parseFromString(this.body, "text/html");
             this.notificationType = htmlBody.querySelector(".o_mail_notification")?.dataset.oeType;
         }
     }
 
-    attachments = Record.many("Attachment");
+    attachments = Record.many("Attachment", { inverse: "message" });
     author = Record.one("Persona");
     /** @type {string} */
     body;
-    composer = Record.one("Composer", { onDelete: (r) => r.delete() });
+    composer = Record.one("Composer", { inverse: "message", onDelete: (r) => r.delete() });
     /** @type {string} */
-    defaultSubject;
+    default_subject;
     /** @type {number|string} */
     id;
     /** @type {boolean} */
-    isDiscussion;
+    is_discussion;
     /** @type {boolean} */
-    isNote;
+    is_note;
     /** @type {boolean} */
     isStarred;
     /** @type {boolean} */
-    isTransient;
-    linkPreviews = Record.many("LinkPreview");
+    is_transient;
```

Comment:
Yes, I think the choice will be on who has authority on the data, and in most cases it's in the server, therefore it's at the choice of the server code, which is currently Python so likely `snake_case`.

I understand that JS community prefers camelCase and Python community prefers snake_case. I think it's a mistake to enforce that, because this implies name conversion in our codebase, which means lots of boilerplate and hard to maintain code.

The best trade-off at the moment is keep a single representation and make the authority on data choose the naming. The formatter had `is_transient` and the server has authority, so I decided `is_transient` in this diff. Python formatter could provide `isTransient`, honestly I don't care much as long as we don't have conversions to do, as long as we pick one for now and we progress faster with code cleaning :)